tls/s2n_server_hello_retry.c

```diff
@@ -32,12 +32,17 @@ uint8_t hello_retry_req_random[S2N_TLS_RANDOM_DATA_LEN] = {
     0xC2, 0xA2, 0x11, 0x16, 0x7A, 0xBB, 0x8C, 0x5E, 0x07, 0x9E, 0x09, 0xE2, 0xC8, 0xA8, 0x33, 0x9C
 };
 
+bool s2n_server_hello_is_hrr(struct s2n_connection *conn)
+{
+    notnull_check(conn);
+    return (memcmp(hello_retry_req_random, conn->secure.server_random, S2N_TLS_RANDOM_DATA_LEN) == 0);
```

Comment:
This function returns bool type, true if server_random is set to `hello_retry_req_random` and false otherwise. So, using `S2N_SUCCESS` doesn't seem like the right approach here.